examples/bri-3/src/bri/auth/guards/didJwt.guard.ts

```diff
@@ -0,0 +1,73 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { IS_PUBLIC_ENDPOINT_KEY } from 'src/bri/decorators/public-endpoint';
+import { Resolver } from 'did-resolver';
+import { getResolver } from 'ethr-did-resolver';
+import { verifyJWT } from 'did-jwt';
+
+@Injectable()
+export class DidJwtAuthGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const isPublic = this.reflector.getAllAndOverride<boolean>(
+      IS_PUBLIC_ENDPOINT_KEY,
+      [context.getHandler(), context.getClass()],
+    );
+    if (isPublic) {
+      return true;
+    }
+    const request = this.getRequest<any>(context);
+    try {
+      const token = this.getToken(request);
+      const verified = await this.verifyJwt(token);
+      console.log('jwt verification result ', verified);
+      return verified.verified;
+    } catch (e) {
+      console.log('jwt verification error ', e);
+      return false;
+    }
+  }
+
+  private async getDidResolver() {
+    const rpcUrl = process.env.GOERLI_RPC_URL;
+
+    const providerConfig = {
+      networks: [{ name: '0x5', rpcUrl }],
```

Comment:
would be better to move to a constant in order to save efforts if this changes anytime down the line